server/TypeDBServer.java

```diff
@@ -104,7 +104,10 @@ protected TypeDBServer(CoreConfig config, boolean debug, Factory factory, CoreLo
         databaseMgr = factory.databaseManager(options);
         server = rpcServer();
         Thread.setDefaultUncaughtExceptionHandler(
-                (t, e) -> logger().error(UNCAUGHT_EXCEPTION.message(t.getName() + ": " + e.getMessage()), e)
+                (t, e) -> {
```

Comment:
Rather than killing the process and exiting with code 1, it would be more elegant if we can simply propagate the uncaught exception to the main thread, and let the JVM exit in the most appropriate way.

Would it be possible to do so, @flyingsilverfin ?

Reply:
you can't propagate from this handler to the main thread

Reply:
we decided just to delete the handler

Reply:
Why not exit the whole server gracefully, @flyingsilverfin ? Log the error, print the stack trace, then gracefully shutdown? Surely that has to be the better way then crash right there and then?